neo/test/rawiotest/test_intanrawio.py

```diff
@@ -66,6 +67,24 @@ def test_annotations(self):
         )
         np.testing.assert_array_equal(signal_array_annotations["board_stream_num"][:10], [0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
 
+    
+    def test_correct_reading_one_file_per_channel(self):
+        "Issue: https://github.com/NeuralEnsemble/python-neo/issues/1599"
+        # Test reading of one-file-per-channel format file. The channels should match the raw files
+        file_path = Path(self.get_local_path("intan/intan_fpc_test_231117_052630/info.rhd"))
+        intan_reader = IntanRawIO(filename=file_path)
+        intan_reader.parse_header()
+
+        # This should be the folder where the files of all the channels are stored
+        folder_path = file_path.parent
+        amplifier_file_paths = [p for p in folder_path.iterdir() if "amp" in p.name]
+        
+        channel_names = [p.name[4:-4] for p in amplifier_file_paths]
+
+        for channel_name, amplifier_file_path in zip(channel_names, amplifier_file_paths):
+            data_raw = np.fromfile(amplifier_file_path, dtype=np.int16).squeeze()
+            data_from_neo = intan_reader.get_analogsignal_chunk(channel_ids=[channel_name], stream_index=0).squeeze()
+            np.testing.assert_allclose(data_raw, data_from_neo)
```

Comment:
Could we get a note in comment about why this works...

so p.name[4:-4] means nothing to someone who doesn't know the file format. So I would explain why 4:-4. And if you're feeling ambitious maybe a better name than p (since I know a certain developer who hates one letter variable names....).

Reply:
Good idea. I was lazy here.